src/transformers/modeling_utils.py

```diff
@@ -1064,6 +1064,7 @@ class PreTrainedModel(nn.Module, ModuleUtilsMixin, GenerationMixin, PushToHubMix
 
     is_parallelizable = False
     supports_gradient_checkpointing = False
+    supports_cuda_fp16_gradient_checkpointing = None
```

Comment:
No we are not adding an attribute like this for just one test in the common tester 😅

Reply:
Actually I don't think it's for the common tester only :D I think we should properly throw an exception or a warning to users if they use a model with gradient checkpointing under autocast + fp16 (that I have attempted [here](https://github.com/huggingface/transformers/pull/24247/commits/f158cffd0cb802260d214f026d76a166d837f506) - I've commented it but plan to use something in that logic), the list of models that doesn't support that scenario are listed here: https://github.com/huggingface/transformers/pull/24247#discussion_r1229840709 
Although we can add `supports_gradient_checkpointing=False` I don't think this is a correct behavior since technically it is supported in some cases (autocast + bf16, no autocast...)

Reply:
Hmm I see, what do you think could be the approach here, since this behaviour is very specific to some models and not observed for other architectures, there is probably a much simpler approach 🤔

Reply:
Honestly this is a bug in PyTorch, so we should not test it for the models where it doesn't work (by manually adding the test in each of those test files and skipping it) instead of adding a new model attribute.